contracts/lib/FulfillmentApplier.sol

```diff
@@ -225,262 +234,185 @@ contract FulfillmentApplier is FulfillmentApplicationErrors {
+                    if xor(firstFulfillmentHeadPtr, fulfillmentHeadPtr) {
+                      let firstFulfillmentPtr := mload(firstFulfillmentHeadPtr)
+                      let fulfillmentPtr := mload(fulfillmentHeadPtr)
+                      mstore(firstFulfillmentHeadPtr, fulfillmentPtr)
+                      mstore(fulfillmentHeadPtr, firstFulfillmentPtr)
+                    }
+                }
+                default {
+                    // Compare every subsequent item to the first
+                    if iszero(
+                        and(
+                            and(
+                                // The offerer must match on both items.
+                                eq(
+                                    mload(paramsPtr),
+                                    mload(
+                                        add(execution, Execution_offerer_offset)
+                                    )
+                                ),
+                                // The conduit key must match on both items.
+                                eq(
+                                    mload(
+                                        add(
+                                            paramsPtr,
+                                            OrderParameters_conduit_offset
+                                        )
+                                    ),
+                                    mload(
+                                        add(execution, Execution_conduit_offset)
+                                    )
+                                )
+                            ),
+                            // The itemType, token, and identifier must match.
+                            eq(
+                                dataHash,
+                                keccak256(
+                                    offerItemPtr,
+                                    ReceivedItem_CommonParams_size
+                                )
+                            )
+                        )
+                    ) {
+                        // Throw if any of the requirements are not met.
+                        throwInvalidFulfillmentComponentData()
+                    }
+                }
+
+                // Retrieve amount pointer using consideration item pointer.
+                let amountPtr := add(offerItemPtr, Common_amount_offset)
 
                 // Add offer amount to execution amount.
                 let newAmount := add(amount, mload(amountPtr))
 
                 // Update error buffer: 1 = zero amount, 2 = overflow, 3 = both.
                 errorBuffer := or(
-                  errorBuffer,
-                  or(
-                    shl(1, lt(newAmount, amount)),
-                    iszero(mload(amountPtr))
-                  )
+                    errorBuffer,
+                    or(shl(1, lt(newAmount, amount)), iszero(mload(amountPtr)))
```

Comment:
might be cheaper to cache `mload(amountPtr)` as it has been also used on line 403.

Reply:
had to do this to manage stack pressure on the coverage tests 🙃